src/overlays/actors/ovl_En_Invadepoh_Demo/z_en_invadepoh_demo.c

```diff
@@ -35,134 +105,706 @@ ActorInit En_Invadepoh_Demo_InitVars = {
+    Actor_ProcessInitChain(&this->actor, sAlienInitChain);
+    this->actor.flags = ACTOR_FLAG_10 | ACTOR_FLAG_IGNORE_QUAKE | ACTOR_FLAG_80000000;
+    this->objectIndex = Object_GetIndex(&play->objectCtx, OBJECT_UCH);
+    if (this->objectIndex < 0) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+
+    this->actionFunc = EnInvadepohDemo_Alien_WaitForObject;
+}
+
+void EnInvadepohDemo_Romani_Init(EnInvadepohDemo* this, PlayState* play) {
+    Actor_ProcessInitChain(&this->actor, sRomaniInitChain);
+    this->objectIndex = Object_GetIndex(&play->objectCtx, OBJECT_MA1);
+    if (this->objectIndex < 0) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+
+    this->actionFunc = EnInvadepohDemo_Romani_WaitForObject;
+}
+
+void EnInvadepohDemo_Cow_Init(EnInvadepohDemo* this, PlayState* play) {
+    Actor_ProcessInitChain(&this->actor, sCowInitChain);
+    Actor_SpawnAsChild(&play->actorCtx, &this->actor, play, ACTOR_EN_INVADEPOH_DEMO, 0.0f, 0.0f, 0.0f, 0, 0, 0,
+                       EN_INVADEPOH_DEMO_TYPE_COW_TAIL);
+    this->objectIndex = Object_GetIndex(&play->objectCtx, OBJECT_COW);
+    if (this->objectIndex < 0) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+
+    this->actionFunc = EnInvadepohDemo_Cow_WaitForObject;
+}
+
+void EnInvadepohDemo_Ufo_Init(EnInvadepohDemo* this, PlayState* play) {
+    Actor_ProcessInitChain(&this->actor, sUfoInitChain);
+    this->actionFunc = EnInvadepohDemo_SelectCueAction;
+    this->drawFlags |= DRAW_FLAG_SHOULD_DRAW;
+}
+
+void EnInvadepohDemo_CowTail_Init(EnInvadepohDemo* this, PlayState* play) {
+    Actor_ProcessInitChain(&this->actor, sCowTailInitChain);
+    this->objectIndex = Object_GetIndex(&play->objectCtx, OBJECT_COW);
+    if (this->objectIndex < 0) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+
+    this->actionFunc = EnInvadepohDemo_CowTail_WaitForObject;
+}
+
+void EnInvadepohDemo_Cow_Destroy(EnInvadepohDemo* this, PlayState* play) {
+    Actor* thisx = &this->actor;
+
+    if (thisx->child != NULL) {
+        thisx->child->parent = NULL;
+    }
+
+    Actor_Kill(thisx->child);
+}
+
+void EnInvadepohDemo_CowTail_Destroy(EnInvadepohDemo* this, PlayState* play) {
+    if (this->actor.parent != NULL) {
+        this->actor.parent->child = NULL;
+    }
+}
+
+void EnInvadepohDemo_Alien_Idle(EnInvadepohDemo* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
+}
+
+void EnInvadepohDemo_Alien_FollowPath(EnInvadepohDemo* this, PlayState* play) {
+    Path* path = &play->setupPathList[this->pathIndex];
+    s32 pathCount = path->count;
+    Vec3s* points = Lib_SegmentedToVirtual(path->points);
+    Vec3f point;
+
+    if (pathCount != 0) {
+        Math_Vec3s_ToVec3f(&point, &points[this->currentPointIndex]);
+        if (Math_Vec3f_StepTo(&this->actor.world.pos, &point, this->speed) < this->speed) {
+            this->currentPointIndex++;
+            if (this->currentPointIndex >= pathCount) {
```

Comment:
`z_obj_um.c` just has `pointIndex` instead of `currentPointIndex`. Does the `Current` add a lot here?

Reply:
I thought it might be confusing to have both `pathIndex` and `pointIndex`, but eh, it's fine if you just pay close enough attention.